File: 2713-find-the-divisibility-array-of-a-string/2713-find-the-divisibility-array-of-a-string.cpp
class Solution {
public:
    vector<int> divisibilityArray(string word, int m) {
        vector<int> ans(word.size(), 0);
        long long now = 0;
        for(int i = 0; i < word.size(); i++){
            now = ((now * 10) % m + (word[i] - '0') % m) % m;
            ans[i] = (now == 0);
        }
        return ans;
    }
};

File: 2713-find-the-divisibility-array-of-a-string/2713-find-the-divisibility-array-of-a-string.py
class Solution:
    def divisibilityArray(self, word: str, m: int) -> List[int]:
        ans = []
        now = 0
        for num in list(word):
            now = (now * 10 + int(num)) % m
            ans.append(int(now == 0))
        return ans